# Exercise 35

A mining company has two mines.

One day's operation at mine 1 produces ore that contains:

- 20 metric tons of copper
- 550 kilograms of silver

One day's operation at mine 2 produces ore that contains:

- 30 metric tons of copper
- 500 kilograms of silver


Let:

$$
v_1=
\begin{bmatrix}
20\\
550
\end{bmatrix}
$$


and


$$
v_2=
\begin{bmatrix}
30\\
500
\end{bmatrix}
$$


Then \(v_1\) and \(v_2\) represent the output per day of mine 1 and mine 2.

---

## (a)

What physical interpretation can be given to:

$$
5v_1
$$


---

## (b)

Suppose the company operates mine 1 for \(x_1\) days and mine 2 for \(x_2\) days.

Write a vector equation whose solution gives the number of days each mine should operate in order to produce:

- 150 metric tons of copper
- 2825 kilograms of silver

Do not solve the equation.

---

## (c)

Solve the equation in part (b).

## 35 (a)

We have:

In [4]:
import sympy as sp
v1=sp.Matrix([20,550])
print(5*v1)

Matrix([[100], [2750]])


The vector 5v1 represents the output of mine 1 operating for 5 days.  
During these 5 days, mine 1 produces:  
100 metric tons of copper  
2750 kilograms of silver  

## (b) (c)

In [5]:

import sympy as sp

A = sp.Matrix([
    [20, 30],
    [550, 500]
])

b = sp.Matrix([
    150,
    2825
])

Ab = A.row_join(b)
rref_matrix, pivot_columns = Ab.rref()
print(rref_matrix)

Matrix([[1, 0, 3/2], [0, 1, 4]])


Therefore:  
- x1=1.5  
- x2=4

Mine 1 operates for 1.5 days,and mine 2 operatres for 4 days.

# Exercise 36

A steam plant burns two types of coal:

- anthracite (A)
- bituminous (B)


For each ton of A burned, the plant produces:

- 27.6 million Btu of heat
- 3100 grams of sulfur dioxide
- 250 grams of particulate matter


For each ton of B burned, the plant produces:

- 30.2 million Btu of heat
- 6400 grams of sulfur dioxide
- 360 grams of particulate matter


Let \(x_1\) be tons of A and \(x_2\) be tons of B.

---

## (a)

How much heat does the steam plant produce?


---

## (b)

Express the output as a linear combination of two vectors.


---

## (c)

The plant produced:

- 162 million Btu of heat
- 23610 g sulfur dioxide
- 1623 g particulate matter

Determine how many tons of each type of coal were burned.

# 36 (a)

Anthracite produces:27.6x1 million Btu

Bituminous produces:30.2x2 million Btu

## (b)

In [17]:
import sympy as sp
# For anthracite:
A=sp.Matrix([27.6,3100,250])
# For bituminous:
B=sp.Matrix([30.2,6400,360])

The total output is:  
x1*A+x2*B

## (C)

In [8]:

import sympy as sp

A = sp.Matrix([
    [27.6, 30.2, 162],
    [3100, 6400, 23610],
    [250, 360, 1623]
])

rref_matrix, pivot_columns = A.rref()
print(rref_matrix)

Matrix([[1, 0, 39/10], [0, 1, 9/5], [0, 0, 0]])


Therefore:  
Anthracite burned: 3.9 tons  
Bituminous coal burned: 1.8 tons  

# Exercises 47-50

Determine if the columns of the matrix span $ R^4 $.

---

## 47.

$$
A=
\begin{bmatrix}
7&2&-5&8\\
-5&-3&4&-9\\
6&10&-2&7\\
-7&9&2&15
\end{bmatrix}
$$


---

## 48.

$$
A=
\begin{bmatrix}
5&-7&-4&9\\
6&-8&-7&5\\
4&-4&-9&-9\\
-9&11&16&7
\end{bmatrix}
$$


---

## 49.

$$
A=
\begin{bmatrix}
12&-7&11&-9&5\\
-9&4&-8&7&-3\\
-6&11&-7&3&-9\\
4&-6&10&-5&12
\end{bmatrix}
$$


---

## 50.

$$
A=
\begin{bmatrix}
8&11&-6&-7&13\\
-7&-8&5&6&-9\\
11&7&-7&-9&-6\\
-3&4&1&8&7
\end{bmatrix}
$$

In [10]:

import sympy as sp

A47 = sp.Matrix([
    [7, 2, -5, 8],
    [-5, -3, 4, -9],
    [6, 10, -2, 7],
    [-7, 9, 2, 15]
])

A48 = sp.Matrix([
    [5, -7, -4, 9],
    [6, -8, -7, 5],
    [4, -4, -9, -9],
    [-9, 11, 16, 7]
])

A49 = sp.Matrix([
    [12, -7, 11, -9, 5],
    [-9, 4, -8, 7, -3],
    [-6, 11, -7, 3, -9],
    [4, -6, 10, -5, 12]
])


A50 = sp.Matrix([
    [8, 11, -6, -7, 13],
    [-7, -8, 5, 6, -9],
    [11, 7, -7, -9, -6],
    [-3, 4, 1, 8, 7]
])

A47_rref_matrix, A47_pivot_columns = A47.rref()
A48_rref_matrix, A48_pivot_columns = A48.rref()
A49_rref_matrix, A49_pivot_columns = A49.rref()
A50_rref_matrix, A50_pivot_columns = A50.rref()

print(A47_rref_matrix,A47_pivot_columns)
print(A48_rref_matrix,A48_pivot_columns)
print(A49_rref_matrix,A49_pivot_columns)
print(A50_rref_matrix,A50_pivot_columns)

Matrix([[1, 0, 0, -93/50], [0, 1, 0, 53/50], [0, 0, 1, -189/50], [0, 0, 0, 0]]) (0, 1, 2)
Matrix([[1, 0, 0, 4/3], [0, 1, 0, -5/3], [0, 0, 1, 7/3], [0, 0, 0, 0]]) (0, 1, 2)
Matrix([[1, 0, 0, -10/21, 0], [0, 1, 0, -25/84, 0], [0, 0, 1, -41/84, 0], [0, 0, 0, 0, 1]]) (0, 1, 2, 4)
Matrix([[1, 0, -7/13, 0, 0], [0, 1, -2/13, 0, 0], [0, 0, 0, 1, 0], [0, 0, 0, 0, 1]]) (0, 1, 3, 4)


## 47

Row reduction gives only 3 pivot positions.  
There is no pivot in every row, so the columns do not span $ R^4 $.

## 48

Row reduction gives only 3 pivot positions.  
There is no pivot in every row, so the columns do not span $ R^4 $.

## 49

Row reduction gives 4 pivot positions.  
There is a pivot in every row, so the columns span $ R^4$.

## 50

Row reduction gives 4 pivot positions.  
There is a pivot in every row, so the columns span $ R^4$.

# 2.1 33

# Exercise 33

Suppose \(A\) is an $ m \times n $ matrix and there exist $ n \times m $ matrices \(C\) and \(D\) such that:

$$
CA=I_n
$$

and

$$
AD=I_m
$$

Prove that:

$$
m=n
$$

and

$$
C=D
$$

**Hint:** Think about the product:

$$
CAD
$$

Since:  
CA=$ I_n $  
we get:  
CAD=$ I_n $D  
Therefore:  
CAD=D  
Since:  
AD=$ I_m $  
we have:  
CAD=C$ I_m $  
Therefore:  
CAD=C  
Hence:  
C=CAD=D  
so:  
C=D  
we have:  
DA=CA  
But:  
CA=$ I_n $  
so:  
DA=$ I_n $  
Also:  
AD=$ I_m $  
Therefore D is both:  
- a left inverse of A
- a right inverse of A
A matrix has a two-sided inverse only if it is square.  

so:
m=n

# Exercise 48

Describe in words what happens when you compute:

$$
A^5,\ A^{10},\ A^{20},\ A^{30}
$$

for

\[
A=
\begin{bmatrix}
\frac16&\frac12&\frac13\\
\frac12&\frac14&\frac14\\
\frac13&\frac14&\frac5{12}
\end{bmatrix}
\]

In [2]:
import sympy as sp
A=sp.Matrix([[1/6,1/2,1/3],
            [1/2,1/4,1/4],
            [1/3,1/4,5/12]])
A5=A**5
A10=A**10
A20=A**20
A30=A**30
print(A5)
print(A10)
print(A20)
print(A30)

Matrix([[0.331822273662551, 0.334587191358025, 0.333590534979424], [0.334587191358025, 0.332320601851852, 0.333092206790123], [0.333590534979424, 0.333092206790123, 0.333317258230453]])
Matrix([[0.333337254947295, 0.333330106839400, 0.333332638213304], [0.333330106839401, 0.333335989260343, 0.333333903900257], [0.333332638213304, 0.333333903900256, 0.333333457886439]])
Matrix([[0.333333333359606, 0.333333333311714, 0.333333333328680], [0.333333333311714, 0.333333333351123, 0.333333333337162], [0.333333333328680, 0.333333333337162, 0.333333333334157]])
Matrix([[0.333333333333333, 0.333333333333333, 0.333333333333333], [0.333333333333333, 0.333333333333333, 0.333333333333333], [0.333333333333333, 0.333333333333333, 0.333333333333333]])


As the exponent increases,the entries of $ A^n $ become closer to 1/3.The matrices $ A^5 A^{10} A^{20} A^{30} $ approach the same matrix with every entry equal to 1/3.

# Exercise 49

The matrix \(M\) can detect a particular $ 2 \times 2 $ colored pattern like in Example 9.

Create a nonzero $ 4 \times 1$ vector \(x\) by choosing each entry to be a zero or one.

Test to see if \(x\) corresponds to the right pattern by calculating:

$$
x^T M x
$$

If

$$
x^T M x = 0
$$

then \(x\) is the pattern identified by \(M\).

If

$$
x^T M x \neq 0
$$

try a different nonzero vector of zeros and ones.

The matrix is:

$$
M=
\begin{bmatrix}
1&0&-1&0\\
0&1&0&0\\
-1&0&1&0\\
0&0&0&1
\end{bmatrix}
$$

In [5]:

import sympy as sp

M = sp.Matrix([
    [1, 0, -1, 0],
    [0, 1, 0, 0],
    [-1, 0, 1, 0],
    [0, 0, 0, 1]
])

for x1 in range(2):
    for x2 in range(2):
        for x3 in range(2):
            for x4 in range(2):

                
                x = sp.Matrix([
                    x1,
                    x2,
                    x3,
                    x4
                ])
                if x != sp.zeros(4, 1):
                    value = (x.T * M * x)[0]
                    if value == 0:
                        print(x)

Matrix([[1], [0], [1], [0]])


Therefore, this vector represents the $ 2\times2 $ colored pattern detected by M.

# Exercise 50

Repeat Exercise 49 with the matrix:

$$
M=
\begin{bmatrix}
1&0&0&-1\\
0&1&0&-1\\
0&0&1&0\\
-1&-1&0&2
\end{bmatrix}
$$

Find a nonzero $ 4\times1  $ vector \(x\) whose entries are 0 or 1 such that:

$$
x^TMx=0
$$

The vector \(x\) represents the  $ 2\times2 $ colored pattern detected by \(M\).

In [6]:

import sympy as sp

M = sp.Matrix([
    [1, 0, 0, -1],
    [0, 1, 0, -1],
    [0, 0, 1, 0],
    [-1,-1, 0, 2]
])

for x1 in range(2):
    for x2 in range(2):
        for x3 in range(2):
            for x4 in range(2):

                x = sp.Matrix([
                    x1,
                    x2,
                    x3,
                    x4
                ])
                if x != sp.zeros(4, 1):
                    value = (x.T * M * x)[0]
                    if value == 0:
                        print(x)

Matrix([[1], [1], [0], [1]])


Therefore, this vector represents the $ 2\times2 $ colored pattern detected by M.

# Exercise 51

Use the matrix

$$
A=
\begin{bmatrix}
0&1\\
1&0
\end{bmatrix}
$$

to switch the first and second rows of the matrix \(M\) containing dates of accidents at the Montreal Trudeau Airport.

The matrix is:

$$
M=
\begin{bmatrix}
2&3&16&24&25&26&6&7&19&26\\
1&1&1&1&1&1&2&2&2&2
\end{bmatrix}
$$

Find the matrix:

$$
AM
$$

In [7]:

import sympy as sp
A = sp.Matrix([
    [0, 1],
    [1, 0]
])

M = sp.Matrix([
    [2, 3, 16, 24, 25, 26, 6, 7, 19, 26],
    [1, 1, 1, 1, 1, 1, 2, 2, 2, 2]
])
ans = A * M
print(ans)

Matrix([[1, 1, 1, 1, 1, 1, 2, 2, 2, 2], [2, 3, 16, 24, 25, 26, 6, 7, 19, 26]])


The matrix A acts as a row-switching matrix, exchanging the two rows of M.

# Exercise 52

Use the matrix

$$
B=
\begin{bmatrix}
1&0&0\\
0&1&0
\end{bmatrix}
$$

to remove the last row from the matrix \(N\) containing dates of accidents at the New York JFK Airport.

The matrix is:

$$
N=
\begin{bmatrix}
1&1&1&1&2&2&2\\
1&12&21&22&3&20&21\\
2020&2020&2020&2020&2020&2020&2020
\end{bmatrix}
$$

Find the matrix:

$$
BN
$$

In [8]:

import sympy as sp
B = sp.Matrix([
    [1, 0, 0],
    [0, 1, 0]
])

N = sp.Matrix([
    [1, 1, 1, 1, 2, 2, 2],
    [1, 12, 21, 22, 3, 20, 21],
    [2020, 2020, 2020, 2020, 2020, 2020, 2020]
])

ans = B * N

print(ans)

Matrix([[1, 1, 1, 1, 2, 2, 2], [1, 12, 21, 22, 3, 20, 21]])


Therefore, the last row of N, which contains the year information, is removed.

# Exercise 46

Let

$$
A=
\begin{bmatrix}
-25&-9&-27\\
546&180&537\\
154&50&149
\end{bmatrix}
$$

Find the second and third columns of $ A^{-1} $ without computing the first column.

In [12]:

import sympy as sp


A = sp.Matrix([
    [-25, -9, -27],
    [546, 180, 537],
    [154, 50, 149]
])

e2 = sp.Matrix([
    0,
    1,
    0
])

e3 = sp.Matrix([
    0,
    0,
    1
])

c2 =sp.linsolve((A,e2))

c3 = sp.linsolve((A,e3))

print(c2)

print(c3)

{(3/2, -433/6, 68/3)}
{(-9/2, 439/2, -69)}


so:  
The second column of $ A^{-1} $ is c2  
The third column of $ A^{-1} $ is c3

# Exercises 9-10

Unless otherwise specified, assume that all matrices in these exercises are $n \times n$.

Determine which of the matrices in Exercises 1–10 are invertible.

Use as few calculations as possible. Justify your answers.

---

## 9.

$$
A=
\begin{bmatrix}
4&0&-7&-7\\
-6&1&11&9\\
7&-5&10&19\\
-1&2&3&-1
\end{bmatrix}
$$


---

## 10.

$$
A=
\begin{bmatrix}
5&3&1&7&9\\
6&4&2&8&-8\\
7&5&3&10&9\\
9&6&4&-9&-5\\
8&5&2&11&4
\end{bmatrix}
$$

In [14]:
import sympy as sp
A9 = sp.Matrix([
    [4,0,-7,-7],
    [-6,1,11,9],
    [7,-5,10,19],
    [-1,2,3,-1]
])

A10 = sp.Matrix([
    [5,3,1,7,9],
    [6,4,2,8,-8],
    [7,5,3,10,9],
    [9,6,4,-9,-5],
    [8,5,2,11,4]
])

rref_A9 = A9.rref()
rref_A10 = A10.rref()
print(rref_A9[0])
print(rref_A9[1])
print(rref_A10[0])
print(rref_A10[1])

Matrix([[1, 0, 0, 0], [0, 1, 0, 0], [0, 0, 1, 0], [0, 0, 0, 1]])
(0, 1, 2, 3)
Matrix([[1, 0, 0, 0, 0], [0, 1, 0, 0, 0], [0, 0, 1, 0, 0], [0, 0, 0, 1, 0], [0, 0, 0, 0, 1]])
(0, 1, 2, 3, 4)


A square matrix is invertible if and only if it is row equivalent to the identity matrix.

Therefore:  
A9 and A10 are invertible.

# Exercise 49

Suppose an experiment leads to the following system of equations:

$$
4.5x_1+3.1x_2=19.249
$$

$$
1.6x_1+1.1x_2=6.843
$$

---

## (a)

Solve system (3), and then solve system (4):

$$
4.5x_1+3.1x_2=19.25
$$

$$
1.6x_1+1.1x_2=6.84
$$

In each case, find the exact solution.

---

## (b)

The entries in (4) differ from those in (3) by less than \(0.05\%\).

Find the percentage error when using the solution of (4) as an approximation for the solution of (3).

---

## (c)

Use your matrix program to produce the condition number of the coefficient matrix in (3).

## (a)

In [17]:

import sympy as sp

A = sp.Matrix([
    [4.5, 3.1],
    [1.6, 1.1]
])

b1 = sp.Matrix([
    19.249,
    6.843
])

b2 = sp.Matrix([
    19.25,
    6.84
])

ans1 =sp.linsolve((A,b1))
ans2 =sp.linsolve((A,b2))
print(ans1)
print(ans2)

{(3.94000000000025, 0.489999999999634)}
{(2.89999999999972, 2.0000000000004)}


so,for system(3):    
$ x_1=3.94 $  
$ x_2=0.49 $  

for system(4):  
$ x_1=2.90 $  
$ x_2=2.00 $


## (b)

for x1:  
|2.90-3.94|/3.94×100%=26.40%  
for x2:  
|2.00-0.49|/0.49×100%=308.16%

## (c)

In [22]:

import sympy as sp
A = sp.Matrix([
    [4.5, 3.1],
    [1.6, 1.1]
])

A_inverse = A.inv()

norm_A = A.norm(1)

norm_A_inverse = A_inverse.norm(1)

condition_number = norm_A * norm_A_inverse

print(condition_number)

4636.00000000020


The matrix is sensitive to small errors in the data.

# Exercise 50

Find the condition number of the matrix \(A\) in Exercise 9.

Construct a random vector \(x\in\mathbb{R}^4\) and compute:

$$
b=Ax
$$

Then use your matrix program to compute the solution \(x_1\) of:

$$
Ax=b
$$

To how many digits do \(x\) and \(x_1\) agree?

Find out the number of digits your matrix program stores accurately, and report how many digits of accuracy are lost when \(x_1\) is used in place of the exact solution \(x\).

In [41]:
import numpy as np  

A = np.array([[4, 0, -7, -7], 
              [-6, 1, 11, 9],  
              [7, -5, 10, 19], 
              [-1, 2, 3, -1]], dtype=float) 

condition_number = np.linalg.cond(A)  

print(condition_number)  

x=np.array([1,2,3,4],dtype=float)

b = A @ x  

x1 = np.linalg.solve(A, b) 

print(x)  

print(b)  

print(x1)  

error = np.abs(x - x1)

print(error)  

23682.9278746812
[1. 2. 3. 4.]
[-45.  65. 103.   8.]
[1. 2. 3. 4.]
[6.76791956e-13 1.95075067e-11 9.48530143e-12 9.87654403e-12]


$log_{10}(23682.9) $ =4.37   
so the expected loss of accuracy is about:4 

Python uses double precision floating-point numbers, which normally store about:
15,16 significant digits.

The numerical experiment shows:
- x and x_1 agree to about 12 digits;
- the condition number predicts a loss of about 4 digits;
- the result is consistent with the theory of conditioning of linear systems.

# Exercise 51

Repeat Exercise 50 for the matrix in Exercise 10.

The matrix is:

$$
A=
\begin{bmatrix}
5 & 3 & 1 & 7 & 9\\
6 & 4 & 2 & 8 & -8\\
7 & 5 & 3 & 10 & 9\\
9 & 6 & 4 & -9 & -5\\
8 & 5 & 2 & 11 & 4
\end{bmatrix}
$$

## Tasks

1. Find the condition number of the matrix $A$.

2. Construct a vector:

$$
x\in \mathbb{R}^{5}
$$

3. Compute:

$$
b=Ax
$$

4. Use a matrix program to solve the system:

$$
Ax_1=b
$$

5. Compare the computed solution $x_1$ with the original vector $x$.

6. Determine:
   - How many digits of $x$ and $x_1$ agree.
   - How many digits of accuracy are lost due to the condition number of $A$.


In [44]:
import numpy as np  

A = np.array([[5,3,1,7,9],      
              [6,4,2,8,-8],    
              [7,5,3,10,9],    
              [9,6,4,-9,-5],   
              [8,5,2,11,4]],    
             dtype=float)    

condition_number = np.linalg.cond(A)  

print(condition_number)  

x=np.array([1,2,3,4,5],dtype=float)

b = A @ x  

x1 = np.linalg.solve(A, b) 

print(x)  

print(b)  

print(x1)  

error = np.abs(x - x1)

print(error)  

68622.45483789868
[1. 2. 3. 4. 5.]
[ 87.  12. 111. -28.  88.]
[1. 2. 3. 4. 5.]
[1.35977896e-11 2.89628321e-11 1.43431933e-11 6.75015599e-13
 1.95399252e-14]


$log_{10}(68622) $ =4.84   
so the expected loss of accuracy is about:5

Python uses double precision floating-point numbers, which normally store about:
15,16 significant digits.

The numerical experiment shows:
- x and x_1 agree to about 11 digits;
- the condition number predicts a loss of about 5 digits;
- the result is consistent with the theory of conditioning of linear systems.

# Exercise 52

Solve the equation:

$$
Ax=b
$$

for a suitable vector $b$ to find the last column of the inverse of the fifth-order Hilbert matrix.

The Hilbert matrix is:

$$
A=
\begin{bmatrix}
1&1/2&1/3&1/4&1/5\\
1/2&1/3&1/4&1/5&1/6\\
1/3&1/4&1/5&1/6&1/7\\
1/4&1/5&1/6&1/7&1/8\\
1/5&1/6&1/7&1/8&1/9
\end{bmatrix}
$$


Find:

1. A suitable vector $b$.
2. The solution vector $x$.
3. The condition number of $A$.
4. How many digits in each entry of $x$ are expected to be correct.

The exact solution is:

$$
x=(630,-12600,56700,-88200,44100)^T
$$

In [48]:
import numpy as np  # Import numpy

A = np.array([[1,1/2,1/3,1/4,1/5],
              [1/2,1/3,1/4,1/5,1/6],
              [1/3,1/4,1/5,1/6,1/7],
              [1/4,1/5,1/6,1/7,1/8],
              [1/5,1/6,1/7,1/8,1/9]], dtype=float)

b = np.array([0,0,0,0,1], dtype=float)  

x = np.linalg.solve(A,b)  

print(x)

condition_number = np.linalg.cond(A)  

print(condition_number)

x1=np.array([630,-12600,56700,-88200,44100])
error=abs(x-x1)
print(error)

[   630. -12600.  56700. -88200.  44100.]
476607.25024172297
[2.25554686e-10 2.07000994e-09 3.33966454e-09 6.25732355e-10
 2.29920261e-09]


Therefore:  
$ log_{10}(476607) $=5.68  
The computation may lose about: 5-6 significant digits.  
Using double precision arithmetic, approximately: 9 significant digits remain accurate.


# Exercise 53

Some matrix programs, such as MATLAB, have a command to create Hilbert matrices of various sizes.

If possible, use an inverse command to compute the inverse of a twelfth-order or larger Hilbert matrix:

$$
A^{-1}
$$

Compute:

$$
AA^{-1}
$$

and report what you find.



In [49]:
import numpy as np  

n = 12  

A = np.zeros((n, n))  

for i in range(n):  

    for j in range(n): 

        A[i, j] = 1 / (i + j + 1)  

A_inverse = np.linalg.inv(A)  

result = A @ A_inverse 

print(result) 

condition_number = np.linalg.cond(A)  

print(condition_number)  

[[ 1.00000000e+00 -1.38988687e-06  2.22292797e-05 -5.73797025e-04
   3.36765961e-03 -2.08099757e-02 -9.33545321e-02 -7.61099262e-05
  -8.83284061e-02 -1.30105432e-02  3.86807706e-02 -2.11758193e-04]
 [-7.40616500e-09  9.99999435e-01 -9.00188374e-06 -1.67550493e-04
   2.18736469e-03 -1.45948782e-02 -3.19155281e-02 -3.38527353e-03
  -9.96559905e-02  1.08078091e-02  1.73295291e-02  1.20959170e-03]
 [-3.40065662e-09 -7.57069190e-07  1.00000529e+00 -2.94551677e-04
   2.73833714e-03 -1.31984501e-02 -3.58572159e-04 -3.21522907e-02
  -4.55761338e-02 -3.06037634e-03  2.49239194e-02  7.25301013e-04]
 [ 3.47383609e-10 -5.28316882e-07  1.87093528e-06  9.99580065e-01
   1.99189130e-03 -1.63690264e-02  5.15721858e-03 -3.67339720e-02
  -3.48531812e-02 -5.72710865e-03  2.10174041e-02  6.47648630e-06]
 [-1.16415322e-09 -3.42726707e-07  5.24520874e-06 -3.50952148e-04
   1.00219727e+00 -1.26953125e-02  2.92968750e-03 -5.27343750e-02
  -2.34375000e-02  9.76562500e-04  1.75781250e-02 -1.70898438e-03]
 [-2.

The experiment shows:
- A 12th-order Hilbert matrix is extremely ill-conditioned.
- Computing its inverse using standard floating-point arithmetic is unreliable.
- The product $ AA^{-1}$significantly deviates from the identity matrix.
- Therefore:
Large Hilbert matrices cannot be inverted accurately using ordinary floating-point arithmetic.